testsuite/MDAnalysisTests/lib/test_util.py

```diff
@@ -767,6 +772,70 @@ def test_multiframe_nonsense(self):
         assert_raises(ValueError, mda.coordinates.core.get_writer_for,
                       filename='this.gro', multiframe='sandwich')
 
+    @staticmethod
+    def _check_singleframe(fmt, cls):
+        assert_equal(mda.coordinates.core.get_writer_for('this', format=fmt, multiframe=False),
+                     cls)
+
+    @staticmethod
+    def _check_singleframe_fails(fmt):
+        assert_raises(TypeError,
+                      mda.coordinates.core.get_writer_for,
+                      'this', format=fmt, multiframe=False)
+
+    @staticmethod
+    def _check_multiframe(fmt, cls):
+        assert_equal(mda.coordinates.core.get_writer_for('this', format=fmt, multiframe=True),
+                     cls)
+
+    @staticmethod
+    def _check_multiframe_fails(fmt):
+        assert_raises(TypeError,
+                      mda.coordinates.core.get_writer_for,
+                      'this', format=fmt, multiframe=True)
+
+    formats = [
+        # format name, related class, singleframe, multiframe
+        ('CRD', mda.coordinates.CRD.CRDWriter, True, False),
+        #('ENT', mda.coordinates.PDB.PDBWriter, True, False),
+        ('GRO', mda.coordinates.GRO.GROWriter, True, False),
+        ('MOL2', mda.coordinates.MOL2.MOL2Writer, True, True),
+        ('NCDF', mda.coordinates.TRJ.NCDFWriter, True, True),
+        ('NULL', mda.coordinates.null.NullWriter, True, True),
+        # ('PDB', mda.coordinates.PDB.PDBWriter, True, True), special case, done separately
+        ('PDBQT', mda.coordinates.PDBQT.PDBQTWriter, True, False),
+        ('PQR', mda.coordinates.PQR.PQRWriter, True, False),
+        ('TRR', mda.coordinates.TRR.TRRWriter, True, True),
+        ('XTC', mda.coordinates.XTC.XTCWriter, True, True),
+        ('XYZ', mda.coordinates.XYZ.XYZWriter, True, True),
+    ]
+    if six.PY2:
```

Comment:
It would be better to test if the DCD writer exists rather than checking the python version. This way, there is no risk to forget to activate the test on python 3 when DCD will work.